test/cctest/test_inspector_socket.cc

```diff
@@ -0,0 +1,849 @@
+#include "inspector_socket.h"
+
+#include "gtest/gtest.h"
+
+#define PORT 9444
+
+static const int MAX_LOOP_ITERATIONS = 10000;
+
+#define SPIN_WHILE(condition)                                                  \
+  {                                                                            \
+    bool timed_out = false;                                                    \
+    timeout_timer.data = &timed_out;                                           \
+    uv_timer_start(&timeout_timer, set_timeout_flag, 5000, 0);                 \
+    while (((condition)) && !timed_out) {                                      \
+      uv_run(&loop, UV_RUN_NOWAIT);                                            \
+    }                                                                          \
+    ASSERT_FALSE((condition));                                                 \
+    uv_timer_stop(&timeout_timer);                                             \
+  }
+
+static uv_timer_t timeout_timer;
+static bool connected = false;
+static bool inspector_ready = false;
+static int handshake_events = 0;
+static enum inspector_handshake_event last_event = kInspectorHandshakeHttpGet;
+static uv_loop_t loop;
+static uv_tcp_t server, client_socket;
+static inspector_socket_t inspector;
+static char last_path[100];
+static void (*handshake_delegate)(enum inspector_handshake_event state,
+                                  const char *path, bool *should_continue);
+
+struct read_expects {
+  const char *expected;
+  size_t expected_len;
+  size_t pos;
+  bool read_expected;
+  bool callback_called;
+};
+
+static const char HANDSHAKE_REQ[] = "GET /ws/path HTTP/1.1\r\n"
+                                    "Host: localhost:9222\r\n"
+                                    "Upgrade: websocket\r\n"
+                                    "Connection: Upgrade\r\n"
+                                    "Sec-WebSocket-Key: aaa==\r\n"
+                                    "Sec-WebSocket-Version: 13\r\n\r\n";
+
+static void set_timeout_flag(uv_timer_t* timer) {
+  *((bool*)timer->data) = true;
+}
+
+static void stop_if_stop_path(enum inspector_handshake_event state,
+                              const char *path, bool *cont) {
```

Comment:
Style issue: star leans to the left, e.g. `bool* cont`.  Applies beyond just this line.